crypto/fipsmodule/rsa/rsa.c

```diff
@@ -743,6 +739,15 @@ int RSA_check_key(const RSA *key) {
     return 0;
   }
 
+  if (!rsa_check_public_key(key, key_enc_type)) {
+    return 0;
+  }
+
+  if (key_enc_type == RSA_STRIPPED_KEY) {
+    // stripped keys doesn't have more parameters we can verify.
+    return 1;
+  }
+
   if (key->d == NULL || key->p == NULL) {
```

Comment:
Why doesn't this look at `RSA_PUBLIC_KEY` instead of looking at the contained parameters?

Reply:
one goal of this change (per guidance from the AWS-LC team earlier in this PR) is to make this change as minimal as possible. so, we tried to preserve original logic inasmuch as possible to avoid accidentally modifying the behavior beyond ACCP's immediate need. prior iterations of this PR did attempt to use the new "key type" enum more liberally, but we walked that back a bit. in the current revision of this PR, the ACCP detector logic doesn't determine whether  to set `RSA_PUBLIC_KEY` as the key type -- that's only done explicitly in certain parts of the code.

this is kind of a long-winded way of saying "we could do that, but it's out of scope because consolidating our key validation logic needs to be done as a standalone paroject with proper design process". I'm currently working on a SIM with some suggestions for how AWS-LC might be able to do that.